tests/valid_basics.stoffel

```diff
@@ -0,0 +1,26 @@
+# This is a single-line comment
+
+# Literals
+let an_int = 42
+let a_float = 123.456 # Fixed point representation
+let a_string = "Hello, Stoffel!"
+let is_true = true
+let is_false = false
+let nothing = nil
+
+# Variable Declarations
+let immutable_val = 100 # Inferred type
```

Comment:
I don't think inferred types are a good idea. You want to be able to know exactly the type and how it affects performance in the Stoffel VM.